### ISSUE TITLE
atomWithObservable can't recover from error (and also with any atoms?)

### ISSUE THREAD

**Issue by dai-shi** [pmndrs/jotai#601]
While working on #599, I noticed retrying from ErrorBoundary doesn't work (there's a failing test skipped). It's probably because the atom value (error) is stored in Provider, so re-render with the atom doesn't trigger a retry. If this is the case, we'd need to find a fundamental solution. 🤔 

Better to start with investigating how useState + useContext would work.

**Comment by tsanyqudsi** [comment]
jotai v.1.2.1 apparently have this :

```
node_modules/jotai/utils/atomWithObservable.d.ts:13:5 - error TS1170: A computed property name in a type literal must refer to an expression whose type is a literal type or a 'unique symbol' type.

13     [Symbol.observable]?: () => ObservableLike<T>;
       ~~~~~~~~~~~~~~~~~~~

node_modules/jotai/utils/atomWithObservable.d.ts:13:13 - error TS2339: Property 'observable' does not exist on type 'SymbolConstructor'.

13     [Symbol.observable]?: () => ObservableLike<T>;
```

**Comment by dai-shi** [comment]
@tsanyqudsi Sorry about the inconvenience, #605 will fix it.

**Comment by tsanyqudsi** [comment]
my problem is confirmed : fixed with #605

**Comment by dai-shi** [comment]
@tsanyqudsi v1.2.2 is published with #605.

**Comment by tsanyqudsi** [comment]
updated and fixed. Thanks @dai-shi :D
